### PR TITLE
[develop < T0021-GA] Add batch save methods

### LINKED ISSUE #64: Add batch save method for saving multiple objects at once
Add a method for saving multiple objects at once. 
For example, currently, you have to iterate over a list of objects and save each of them separately to Memgraph.
 
```python
db.save_all(nodes_list)
```

### PR DESCRIPTION
**Pull request by g-despot**
Add methods for batch saving nodes and relationships to the database:

- [x] `save_nodes()`
- [x] `save_relationships()`
- [x] Create tests for saving multiple nodes and relationship. 

Closes #64

### CODE REVIEW

File: gqlalchemy/memgraph.py [hunk 1]
```diff
@@ -327,6 +327,12 @@ def save_node(self, node: Node) -> Node:
         result = self._save_node_properties_on_disk(node, result)
         return result
 
+    def save_nodes(self, nodes: List[Node]) -> None:
+        """Saves a list of nodes to Memgraph."""
+        for i in range(len(nodes)):
+            saved_node = self.save_node(nodes[i])
+            nodes[i]._id = saved_node._id
+
```
Comment (katarinasupe) [inline gqlalchemy/memgraph.py:335]: Method [`save_node()`](https://github.com/memgraph/gqlalchemy/blob/ad9a4055698928c51de405cf20228e52b158f48a/gqlalchemy/memgraph.py#L301) returns saved node. Here nothing is returned. That's fine if we decide to do it like that, but it's a bit inconsistent with the `save_node()` method.

File: gqlalchemy/memgraph.py [hunk 2]
```diff
@@ -489,6 +495,12 @@ def save_relationship(self, relationship: Relationship) -> Optional[Relationship
         result = self._save_relationship_properties_on_disk(relationship, result)
         return result
 
+    def save_relationships(self, relationships: List[Relationship]) -> None:
+        """Saves a list of relationships to Memgraph."""
+        for i in range(len(relationships)):
+            saved_relationship = self.save_relationship(relationships[i])
+            relationships[i]._id = saved_relationship._id
+
```
Comment (katarinasupe) [inline gqlalchemy/memgraph.py:503]: Same comment applies here.

File: tests/ogm/test_serialisation.py [hunk 3]
```diff
@@ -74,13 +71,36 @@ class SimpleRelationship(Relationship, type="SIMPLE_RELATIONSHIP"):
...
+    relationship = SimpleRelationship(_start_node_id=node1._id, _end_node_id=node2._id,)
     assert SimpleRelationship.type == relationship._type
     assert SimpleRelationship.type is not None
     relationship.save(memgraph)
     assert relationship._id is not None
     relationship2 = memgraph.load_relationship(relationship)
     assert relationship2._id == relationship._id
+
+
+def test_save_relationships(memgraph):
```
Comment (katarinasupe) [inline tests/ogm/test_serialisation.py:83]: I would create new test that tests only saving nodes and name it like that.

**Review by katarinasupe — APPROVED** [review]

File: gqlalchemy/memgraph.py [hunk 4]
```diff
@@ -489,6 +495,12 @@ def save_relationship(self, relationship: Relationship) -> Optional[Relationship
         result = self._save_relationship_properties_on_disk(relationship, result)
         return result
 
+    def save_relationships(self, relationships: List[Relationship]) -> None:
+        """Saves a list of relationships to Memgraph."""
+        for i in range(len(relationships)):
+            saved_relationship = self.save_relationship(relationships[i])
```
Comment (BorisTasevski) [inline gqlalchemy/memgraph.py:501]: Is this line necessary?

File: tests/ogm/test_serialisation.py [hunk 5]
```diff
@@ -44,6 +44,36 @@ class NodeWithKey(Node):
     assert node1.name == node2.name
 
 
+def test_save_nodes(memgraph):
+    class SimpleNode(Node):
```
Comment (BorisTasevski) [inline tests/ogm/test_serialisation.py:48]: Class to fixture?

File: tests/ogm/test_serialisation.py [hunk 6]
```diff
@@ -84,3 +114,35 @@ class SimpleRelationship(Relationship, type="SIMPLE_RELATIONSHIP"):
     assert relationship._id is not None
     relationship2 = memgraph.load_relationship(relationship)
     assert relationship2._id == relationship._id
+
+
+def test_save_relationships(memgraph):
+    class User(Node):
```
Comment (BorisTasevski) [inline tests/ogm/test_serialisation.py:120]: Class to fixture?

File: tests/ogm/test_serialisation.py [hunk 7]
```diff
@@ -84,3 +114,35 @@ class SimpleRelationship(Relationship, type="SIMPLE_RELATIONSHIP"):
...
+        id: int = Field(exists=True, unique=True, index=True, db=memgraph)
+        name: Optional[str] = Field()
+
+    class Follows(Relationship, type="FOLLOWS"):
```
Comment (BorisTasevski) [inline tests/ogm/test_serialisation.py:124]: Class to fixture?

**Review by BorisTasevski — CHANGES_REQUESTED** [review]

File: gqlalchemy/memgraph.py (hunk 4 above)
Comment (Josipmrden) [reply gqlalchemy/memgraph.py:501]: The one above or below?

Comment (g-despot) [reply gqlalchemy/memgraph.py]: It was the one below. I removed the unnecessary variable assignment.

**Review by BorisTasevski — APPROVED** [review]

**Pull request merged**